Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SpawnerComponent.h"
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI.h>
+#include <ROS2/ROS2Bus.h>
+
+namespace ROS2
+{
+    ROS2SpawnerComponent::ROS2SpawnerComponent()
+    {
+        SpawnerInterface::Register(this);
```

Comment:
Perhaps this should happen on Activate? Let me know what is the reasoning behind this in constructor.